aw-models/src/event.rs

```diff
@@ -34,6 +35,9 @@ impl Event {
         self.timestamp
             + chrono::Duration::nanoseconds(self.duration.num_nanoseconds().unwrap() as i64)
     }
+    pub fn interval(&self) -> TimeInterval {
+        TimeInterval::new(self.timestamp, self.calculate_endtime())
```

Comment:
I still have not found out why, but it calculating endtime is incredibly slow.
It shouldn't be a problem right now, just something that I found out when trying out "yearly" view and benchmarking. Was able to speed up some queries by 50% by reducing the amount of calculate_endtime().